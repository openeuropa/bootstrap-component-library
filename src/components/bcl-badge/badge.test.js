import { renderTwigFileAsNode, getVariants } from "@openeuropa/bcl-test-utils";
import demoData from "@openeuropa/bcl-data-badge/data";

const template = "@oe-bcl/bcl-badge/badge.html.twig";
const render = (params) => renderTwigFileAsNode(template, params, true);
const backgrounds = getVariants();

describe("OE - badge", () => {
  test("renders correctly with a link", () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, url: "http://example.com" })
    ).resolves.toMatchSnapshot();
  });
  test("renders correctly with assistive text", () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, assistive_text: "An assistive text" })
    ).resolves.toMatchSnapshot();
  });

  test("renders correctly as a rounded pill", () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, rounded_pill: true })
    ).resolves.toMatchSnapshot();
  });

  test("renders correctly dismissible", () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, dismissible: true })
    ).resolves.toMatchSnapshot();
  });

  test("renders correctly with title", () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, url: "http://example.com", title: "new title" })
    ).resolves.toMatchSnapshot();
  });

  backgrounds.forEach((background) => {
    test(`${background} renders correctly`, () => {
      expect.assertions(1);

      return expect(
        render({ ...demoData, background })
      ).resolves.toMatchSnapshot();
    });

    test(`${background} outline renders correctly`, () => {
      expect.assertions(1);

      return expect(
        render({ ...demoData, background, outline: true })
      ).resolves.toMatchSnapshot();
    });
  });
});
