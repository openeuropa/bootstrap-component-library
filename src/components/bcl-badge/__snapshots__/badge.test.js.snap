// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OE - badge danger renders correctly 1`] = `
<jest>
  <span
    class="badge bg-danger"
  >
    A simple badge!
  </span>
</jest>
`;

exports[`OE - badge dark renders correctly 1`] = `
<jest>
  <span
    class="badge bg-dark"
  >
    A simple badge!
  </span>
</jest>
`;

exports[`OE - badge info renders correctly 1`] = `
<jest>
  <span
    class="badge bg-info text-dark"
  >
    A simple badge!
  </span>
</jest>
`;

exports[`OE - badge light renders correctly 1`] = `
<jest>
  <span
    class="badge bg-light text-dark"
  >
    A simple badge!
  </span>
</jest>
`;

exports[`OE - badge primary renders correctly 1`] = `
<jest>
  <span
    class="badge bg-primary"
  >
    A simple badge!
  </span>
</jest>
`;

exports[`OE - badge renders correctly as a rounded pill 1`] = `
<jest>
  <span
    class="badge bg-primary rounded-pill"
  >
    A simple badge!
  </span>
</jest>
`;

exports[`OE - badge renders correctly with a link 1`] = `
<jest>
  <a
    href="http://example.com"
    title="A simple badge!"
  >
    <span
      class="badge bg-primary"
    >
      A simple badge!
    </span>
  </a>
</jest>
`;

exports[`OE - badge renders correctly with assistive text 1`] = `
<jest>
  <span
    class="badge bg-primary"
  >
    <span
      class="visually-hidden"
    >
      An assistive text
    </span>
    A simple badge!
  </span>
</jest>
`;

exports[`OE - badge secondary renders correctly 1`] = `
<jest>
  <span
    class="badge bg-secondary"
  >
    A simple badge!
  </span>
</jest>
`;

exports[`OE - badge success renders correctly 1`] = `
<jest>
  <span
    class="badge bg-success"
  >
    A simple badge!
  </span>
</jest>
`;

exports[`OE - badge warning renders correctly 1`] = `
<jest>
  <span
    class="badge bg-warning text-dark"
  >
    A simple badge!
  </span>
</jest>
`;
